### ISSUE TITLE
jsdoc -X fails in Node v23 ("util.isRegExp is not a function")

### ISSUE THREAD

**Issue by 75lb** [jsdoc/jsdoc#2126]
In NodeJs version 23, running this command against any file...

```
$ jsdoc -X index.js
```

...fails with this output 

```
jsdoc-issue lloyd:$ npx jsdoc -X index.js
(node:67669) ExperimentalWarning: Support for loading ES Module in require() is an experimental feature and might change at any time
(Use `node --trace-warnings ...` to show where the warning was created)
/Users/lloyd/Documents/jsdoc2md/tmp/jsdoc-issue/node_modules/jsdoc/lib/jsdoc/util/dumper.js:98
        else if ( util.isRegExp(o) ) {
                       ^

TypeError: util.isRegExp is not a function
    at ObjectWalker.walk (/Users/lloyd/Documents/jsdoc2md/tmp/jsdoc-issue/node_modules/jsdoc/lib/jsdoc/util/dumper.js:98:24)
    at /Users/lloyd/Documents/jsdoc2md/tmp/jsdoc-issue/node_modules/jsdoc/lib/jsdoc/util/dumper.js:90:41
    at Array.forEach (<anonymous>)
    at /Users/lloyd/Documents/jsdoc2md/tmp/jsdoc-issue/node_modules/jsdoc/lib/jsdoc/util/dumper.js:89:21
    at ObjectWalker.checkCircularRefs (/Users/lloyd/Documents/jsdoc2md/tmp/jsdoc-issue/node_modules/jsdoc/lib/jsdoc/util/dumper.js:70:20)
    at ObjectWalker.walk (/Users/lloyd/Documents/jsdoc2md/tmp/jsdoc-issue/node_modules/jsdoc/lib/jsdoc/util/dumper.js:86:27)
    at exports.dump (/Users/lloyd/Documents/jsdoc2md/tmp/jsdoc-issue/node_modules/jsdoc/lib/jsdoc/util/dumper.js:142:44)
    at module.exports.cli.dumpParseResults (/Users/lloyd/Documents/jsdoc2md/tmp/jsdoc-issue/node_modules/jsdoc/cli.js:397:50)
    at module.exports.cli.processParseResults (/Users/lloyd/Documents/jsdoc2md/tmp/jsdoc-issue/node_modules/jsdoc/cli.js:385:17)
    at module.exports.cli.main (/Users/lloyd/Documents/jsdoc2md/tmp/jsdoc-issue/node_modules/jsdoc/cli.js:235:18)
    at module.exports.cli.runCommand (/Users/lloyd/Documents/jsdoc2md/tmp/jsdoc-issue/node_modules/jsdoc/cli.js:186:9)
    at /Users/lloyd/Documents/jsdoc2md/tmp/jsdoc-issue/node_modules/jsdoc/jsdoc.js:93:9
    at Object.<anonymous> (/Users/lloyd/Documents/jsdoc2md/tmp/jsdoc-issue/node_modules/jsdoc/jsdoc.js:94:3)
    at Module._compile (node:internal/modules/cjs/loader:1568:14)
    at Object..js (node:internal/modules/cjs/loader:1711:10)
    at Module.load (node:internal/modules/cjs/loader:1328:32)
```

**Comment by 75lb** [comment]
See the [Node v23 Release Notes](https://github.com/nodejs/node/blob/main/doc/changelogs/CHANGELOG_V23.md#semver-major-commits) for details of the deprecated methods now removed. 

**Please review with urgency** as this bug has killed jsdoc in the latest version of Node.

**Comment by hegemonic** [comment]
Fixed in JSDoc 4.0.4.
